.envrc

```diff
@@ -0,0 +1,30 @@
+if ! find_up .venv/pyvenv.cfg; then
+    echo "ITS NOT there"
+    python3 -m venv .venv
+    echo "NOW IT IS"
+fi
+
+source ./.venv/bin/activate
+
+if ! find_up ./.venv/dummy.txt; then
+    echo "ITS NOT there"
+    pip install .
+    pip install '.[dev]'
+    touch ./.venv/dummy.txt
+    echo "NOW IT IS"
+fi
+
+pre-commit install
+
+# source ./.venv/bin/activate
+
+
+
+#Use direnv allow to run stuff within the envrionment
+# git add, git commit, git push
+
+
+#   if [[ ! -d $VIRTUAL_ENV ]]; then
+#         log_status "no venv found; creating $VIRTUAL_ENV"
+#         "$python" -m venv "$VIRTUAL_ENV"
+#     fi
```

Comment:
Remove dead code and unrelated comments. On the topic of dead code, the point of having VCS is to let you go back in time, there's no point in keeping the code commented as it just adds confusion. Why was it commented out? What does it do that it isn't enabled? Why are there variables or features not present in the current code? The point of comments is to explain why, not how (with very few exceptions) since the code already shows us to do that.